src/System.Management.Automation/engine/Utils.cs

```diff
@@ -650,30 +650,31 @@ internal static string GetRegistryConfigurationPath(string shellID)
 #else
             lock (s_cachedGroupPolicySettings)
             {
-                // Return cached information, if we have it
-                Dictionary<string, object> settings;
-                if ((s_cachedGroupPolicySettings.TryGetValue(settingName, out settings)) &&
```

Comment:
If the key was a Tuple: settingName, preferenceOrder - would that fix this issue?

If it helps, we could make preferenceOrder an enum where we select from the 4 lists we of order we use now.